src/cdk/v2/destinations/heap/procWorkflow.yaml

```diff
@@ -0,0 +1,74 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - name: flattenJson
+    path: ../../../../v0/util
+  - path: ../../bindings/jsontemplate
+
+steps:
+  - name: validateInput
+    template: |
+      $.assert(.message.type, "message Type is not present. Aborting message.");
+       $.assert(.message.type in {{$.EventType.([.IDENTIFY, .TRACK])}}, 
+        "message type " + .message.type + " is not supported");
+  - name: setMessageType
+    template: |
+      $.context.messageType = .message.type.toLowerCase()
+  - name: identifyPayload
+    condition: $.context.messageType === {{$.EventType.IDENTIFY}}
+    template: |
+      $.assert(.message.traits ?? .message.context.traits, "traits are required");
+      $.context.endpoint = 'https://heapanalytics.com/api/add_user_properties';
+      $.context.payload = .message.({
+        identity: .userId ?? .traits.userId ?? .traits.id ??
+                  .context.traits.userId ?? .context.traits.id ??
+                  .anonymousId,
```

Comment:
```suggestion
        identity: .message.({{{{$.getGenericPaths("userIdOnly")}}}}),
```

Reply:
why userIdOnly? Are we not using anonymousId?

Reply:
we need to use .message.({{{{$.getGenericPaths("userId")}}}})

Reply:
done